### PR TITLE
Add Edit(Pencil) link for readmes not in the root

### LINKED ISSUE #163: Detect non-root readmes
When you open the repo root GitHub renders the readme file and we place an edit button in the top-right corner of the rendered readme.

However GitHub renders readmes in directories and we don't detect that and don't place the edit buttons there.

Example: https://github.com/rjsteinert/GitHub-For-Writers-Manual/tree/master/managing-multiple-pages

### PR DESCRIPTION
**Pull request by DrewML**
Closes #163 

Test Cases:
[Depth of 1](https://github.com/rjsteinert/GitHub-For-Writers-Manual/tree/master/managing-multiple-pages)
[Depth of 2](https://github.com/DrewML/tester-tester/tree/master/test/anothertest)

### CODE REVIEW

**Comment by paulmolluzzo** [comment]
I kind of just want to point out some "historical" thing here: When we first put this in were were supporting editing .md files in the non-root, such as in a tree. This PR and #163 goes back on that decision to allow directory and tree md file editing. I'm for this update, but @sindresorhus and I had some back and forth about it here and might have an opinion on it still: https://github.com/sindresorhus/refined-github/pull/41#discussion_r54263032

Barring some objection from @sindresorhus I say LGTM :shipit:

File: extension/content.js [hunk 1]
```diff
@@ -384,7 +389,7 @@ document.addEventListener('DOMContentLoaded', () => {
 				addBlameParentLinks();
 			}
 
-			if (pageDetect.isRepoRoot()) {
+			if (pageDetect.isRepo()) {
 				addReadmeEditButton();
 			}
```
Comment (hkdobrev) [inline extension/content.js]: You are already in `pageDetect.isRepo()` if, no need for this one. However I think we can add a tree check to `pageDetect` and run this `if(pageDetect.isRepoRoot() || pageDetect.isRepoTree())`.

Comment (DrewML) [inline extension/content.js]: Good suggestion - updates pushed.

**Comment by hkdobrev** [comment]
LGTM :shipit:

**Comment by DrewML** [comment]
@sindresorhus Any objections?

**Comment by sindresorhus** [comment]
> I'm for this update, but @sindresorhus and I had some back and forth about it here and might have an opinion on it still: #41 (comment)

Looking back at that discussion now it seems I was totally confused about what we were actually talking about. My mistake.

**Pull request merged**